src/Engine/ProtoCore/Lang/Replication/ReplicationInstruction.cs

```diff
@@ -44,5 +45,22 @@ public override string ToString()
             }
 
         }
+
+        public Boolean Equals(ReplicationInstruction oldOption) {
```

Comment:
this does not seem to actually perform a full equality check... it only returns true for zip replication - you should override Equals like this

https://docs.microsoft.com/en-us/dotnet/api/system.object.equals?view=netframework-4.7.2

use the `override keyword`

Reply:
it's a struct - so @reddyashish you may want to read this:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type

Reply:
@mjkkirschner Do you want me change anything in this Equals method after our discussion on friday or is this ok?

Reply:
hmm, well I guess I am a bit confused why you do not just compare the properties directly? ie Why all the use of Except?

Reply:
Added the comments to code, on why we can use Except to compare the elements in both the lists.